### PR TITLE
Split debug info for all targets

### PR DESCRIPTION
**Pull request by expipiplus1**
Work towards https://github.com/shader-slang/slang/issues/5724

### CODE REVIEW

**Comment by expipiplus1** [comment]
I need to test the release script in my repo first, however this is good to review

**Comment by expipiplus1** [comment]
tested here: https://github.com/shader-slang/slang/releases/tag/untagged-fbd0a2148c1599ac1c74

**Review by csyonghe — COMMENTED** [review]
The packages looks good! We need to verify the macOS release is still working.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -152,32 +149,28 @@ jobs:
...
           triggering_ref=${{ github.ref_name }}
           base=slang-${triggering_ref#v}-${{matrix.os}}-${{matrix.platform}}
+
+          # Move main packages
           mv "$(pwd)/build/dist-${config}/slang.zip" "${base}.zip"
           echo "SLANG_BINARY_ARCHIVE_ZIP=${base}.zip" >> "$GITHUB_OUTPUT"
           mv "$(pwd)/build/dist-${config}/slang.tar.gz" "${base}.tar.gz"
           echo "SLANG_BINARY_ARCHIVE_TAR=${base}.tar.gz" >> "$GITHUB_OUTPUT"
 
-          # For some reason, the binaries packed by cpack for macos is modified
```
[the remark this answers is not in the document]
Comment (expipiplus1) [reply .github/workflows/release.yml:165]: I think the only package we notarize is the `slang-macos-dist.zip` still, but at least on windows cpack isn't modifying the files. I don't have a mac to verify, but I think we could add a check in CI for this

[the remark this answers is not in the document]
Comment (expipiplus1) [reply .github/workflows/release.yml:165]: ok, I reverted this change and will tackle it in a follow on PR

**Comment by expipiplus1** [comment]
Finished and tested here: https://github.com/shader-slang/slang/releases/tag/untagged-4d399a8af46382fd1926

**Comment by csyonghe** [comment]
I noticed that the windows build of slang.dll is now 23MB in size.

In comparison, when I build RelWithDebInfo config locally, the binary is 13MB in size (not embeding stdlib). But the slang-stdlib we write out is 3.8MB, so it should be 16.8MB with stdlib embeded.

When I build with Release config, the binary is 8.9MB in size.

This is telling me that Release is running a very different set of optimizations from RelWithDebInfo. I think what we want is the same optimizations enabled with Release, but with pdbs generated.

**Review by csyonghe — APPROVED** [review]
Looks good, but I think we should consider changing the cmake config to release but output pdb from that config, since relWithDebInfo appears to include less optimizations that results larger binary size.

**Comment by csyonghe** [comment]
For simplicity, I would rather us just delete RelWithDebInfo and add debug symbol generation to Release config, similar to what we have before the recent cmake change.

**Comment by expipiplus1** [comment]
oh weird, I actually checked exactly this on Linux when I did the releaseWithDebugInfo builds and it was OK

```
$ ls build/{Release,RelWithDebInfo,Debug}/lib/libslang.so -l
.rwxr-xr-x e users 21 MB Fri Dec  6 11:34:43 2024  build/Release/lib/libslang.so
.rwxr-xr-x e users 21 MB Fri Dec  6 11:34:59 2024  build/RelWithDebInfo/lib/libslang.so
.rwxr-xr-x e users 41 MB Fri Dec  6 11:35:22 2024  build/Debug/lib/libslang.so
```

I guess it's something Windows specific....

**Comment by expipiplus1** [comment]
> For simplicity, I would rather us just delete RelWithDebInfo and add debug symbol generation to Release config, similar to what we have before the recent cmake change.

But this is precisely what `RelWithDebInfo` is supposed to be? Anything which diverges from being as much as possible a vanilla CMake project just makes it more surprising for downstream developers and OS packagers.

**Comment by expipiplus1** [comment]
looks similar on windows if I force the release build to have debug info...
image

**Comment by csyonghe** [comment]
Slang.dll has always been around 9MB in my memory. I don’t know what caused the inflation in size so greatly.

**Pull request merged**